core/src/mindustry/world/blocks/distribution/BaseConveyor.java

```diff
@@ -0,0 +1,94 @@
+package mindustry.world.blocks.distribution;
+
+import arc.*;
+import arc.math.*;
+import arc.util.*;
+import mindustry.gen.*;
+import mindustry.world.*;
+import arc.graphics.g2d.*;
+import mindustry.graphics.*;
+import mindustry.world.meta.*;
+import mindustry.world.blocks.*;
+import mindustry.entities.type.*;
+import mindustry.entities.traits.BuilderTrait.*;
+
+import static mindustry.Vars.tilesize;
+
```

Comment:
extracted some conveyor code into a base conveyor class so i could use it in this fork more cleanly compared to a sub class of conveyor and/or straight up duplicate code